jf_agent/bb_download.py

```diff
@@ -19,34 +37,82 @@ def _normalize_user(user):
 
 
 def get_all_users(client):
-    print('downloading bitbucket users... ✓')
+    print('downloading bitbucket users... ', end='', flush=True)
+    users = [_normalize_user(user) for user in client.admin.users]
+    print('✓')
+    return users
 
-    return [_normalize_user(user) for user in client.admin.users]
 
+def _normalize_project(api_project, redact_names_and_urls):
+    return {
+        'id': api_project['id'],
+        'login': api_project['key'],
+        'name': (
+            api_project.get('name')
+            if not redact_names_and_urls
+            else _project_redactor.redact_name(api_project.get('name'))
+        ),
+        'url': api_project['links']['self'][0]['href'] if not redact_names_and_urls else None,
+    }
 
-def get_all_projects(client, include_projects, exclude_projects):
-    print('downloading bitbucket projects... ✓')
+
+def get_all_projects(client, include_projects, exclude_projects, redact_names_and_urls):
+    print('downloading bitbucket projects... ', end='', flush=True)
 
     filters = []
     if include_projects:
         filters.append(lambda p: p['key'] in include_projects)
     if exclude_projects:
         filters.append(lambda p: p['key'] not in exclude_projects)
 
-    return [
-        {
-            'id': p['id'],
-            'login': p['key'],
-            'name': p.get('name'),
-            'description': p.get('description'),
-            'url': p['links']['self'][0]['href'],
-        }
+    projects = [
+        (p, _normalize_project(p, redact_names_and_urls))
         for p in client.projects.list()
         if all(filt(p) for filt in filters)
     ]
+    print('✓')
+    return projects
+
+
+def _normalize_repo(api_project, api_repo, redact_names_and_urls):
+    repo = api_repo.get()
```

Comment:
Some of the getter logic matches what we do in `adapters.bitbucket`. I wonder if it's worth factoring it out to avoid needing to update two places in the future.